app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,24 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path unless user
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user 
```

Comment:
This solution isn't 100% right as when you think about it - if there is invalid password, you are not going to save it also not going to inform user about it. Instead of setting it manually, and then saving the instance, you should use `update` method and have slightly  more expanded control flow.

```ruby
if user && user.update(password: params[:password])
  user.remove_token! # this would be a method to update user instance with token being nil.
  flash[:success] = "Your password has been changed. Please sign in"
  redirect_to sign_in_path
elsif user
  @token = user.token
  render :show
else
  redirect_to expired_token_path
end
```